iree/compiler/Codegen/Utils/GPUUtils.cpp

```diff
@@ -65,5 +67,52 @@ std::array<int64_t, 3> getWorkgroupSize(mlir::func::FuncOp funcOp) {
   return workgroupSize;
 }
 
+bool canPerformVectorAccessUsingAllThreads(ArrayRef<int64_t> shape,
```

Comment:
note that those functions are super ad hoc and are likely to change and break in the future. It is fine to re-use but just want to point it out if you had different expectations.

Reply:
SG, thanks for the reminder!